dimod/binary/binary_quadratic_model.py

```diff
@@ -763,6 +764,120 @@ def add_linear_equality_constraint(
                     u, v, 2 * lagrange_multiplier * ubias * vbias)
         self.offset += lagrange_multiplier * constant * constant
 
+    def add_linear_inequality_constraint(
+            self, terms: Iterable[Tuple[Variable, int]],
+            lagrange_multiplier: Bias,
+            label: str,
+            constant: int = 0,
+            lb: int = np.iinfo(np.int64).min,
+            ub: int = 0,
+            cross_zero: bool = False
+    ) -> Iterable[Tuple[Variable, int]]:
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form:
+
+        math:'lb <= \sum_{i,k} a_{i,k} x_{i,k} + constant <= ub'
+        to the binary quadratic model as a quadratic objective.
+        For constraints with fractional coefficients, multiply both sides of the
+        inequality by an appropriate factor of ten to attain or approximate
+        integer coefficients.
+
+        Args:
+            terms (iterable/iterator):
+                An iterable of 2-tuples, (variable, bias), with each tuple
+                constituting a term in :math:`\sum_{i} a_{i} x_{i},
+                with i being the length of the iterable.
+            lagrange_multiplier:
+                A weight or the penalty strength. This value is
+                multiplied by the entire constraint objective and added to the
+                binary quadratic model (it doesn't appear explicitly in the
+                equation above).
+            label:
+                Prefix used to label the slack variables used to create the new
+                objective.
+            constant:
+                The constant value of the constraint.
+            lb:
+                lower bound for the constraint
+            ub:
+                upper bound for the constraint
+            cross_zero:
+                When True, adds zero to the domain of constraint
+
+        Returns:
+            slack_terms:  An iterable of 2-tuples for the new slack variables
+             (variable, int), with each tuple constituting a term in
+             :math:`\sum_{i} b_{i} slack {i},
+             with i being the length of the iterable and b being the coefficient
+             for slack variable.
+        """
+
+        if isinstance(terms, Iterator):
```

Comment:
Can we add a test for this case? Something like
```
bqm.add_linear_inequality_constraint(iter(terms), ...)
```
will do it.

Reply:
added this to one of the existing tests.